_dev/src/store/modules/product-feed/actions.ts

```diff
@@ -130,18 +130,33 @@ export default {
     const productFeedSettings = state.settings;
     const targetCountries = changeCountriesNamesToCodes(getters.GET_TARGET_COUNTRIES);
     const attributeMapping = getDataFromLocalStorage('productFeed-attributeMapping') || {};
+    const deliveryEnabled = productFeedSettings.deliveryDetails.filter((e) => e.enabledCarrier);
```

Comment:
You filter one first time the carriers on this line.

All the concerned carriers won't be removed from the `shipping` array.